content/blog/2024/zoomies.md

```diff
@@ -0,0 +1,464 @@
+by the operating system,
+before the browser gets involved.
+If I also zoom a web page by `2x`/`200%` in the browser,
+the result is a `4x`/`400%` overall zoom --
+and a _device pixel ratio_ of `4:1`.
+
+This _zoom_ is applied to the size of a CSS pixel,
+_before the page is rendered_.
+By zooming in,
+we make each 'pixel' larger.
+But our layout viewport isn't growing at all,
+so our layout now contains _fewer_ `px`
+in each dimension.
+In effect, we've made
+the layout viewport _smaller_
+in relation to our pixels.
+
+**Page Zoom** is adjusting
+_the size of a CSS pixel in relation to the layout viewport_.
+Since that happens before rendering,
+it impacts the layout of the page.
+It's then reflected by media queries,
+which query a 'smaller' viewport.
+
+As far as the browser is concerned,
+there's very little difference
+between _making the window smaller_
+or _making the pixels bigger_.
+The result is the same:
+_fewer pixels fit in the viewport_.
+
+### Scale factor: viewport vs. viewport
+
+The _scale factor_
+is also available in all browsers,
+but you're most likely to notice it
+on touch-screen devices.
+As far as I can tell,
+this was implemented originally
+for mobile Safari,
+and later added to the spec,
+and adopted by desktop browsers.
+In fact, the published spec
+mentions a previous name for it --
+_pinch zoom_ --
+and the Editor's Draft
+includes a brief explanation:
+
+> The "scale factor" is often referred to as "pinch-zoom";
+> however, it can be affected through means other than pinch-zooming.
+> e.g. The user agent may zooms [sic] in
+> on a focused input element to make it legible.
+
+I know I've experienced that.
+On mobile Safari you can also double-tap
+elements in the page to 'zoom in'
+so that element fills the viewport.
+Testing here on a MacBook laptop with a track-pad,
+both the pinch and double-tap interactions
+work for me now in macOS Vivaldi (Chromium).
+
+If you play with this,
+you'll notice that it's quite different
+from the behavior of page zoom above.
+First: _we can only zoom in, not out_.
+There is no way to scale the page
+so that it is smaller than 100%
+of the visual viewport.
+And when we do 'scale' the page up,
+_the layout doesn't change,
+but we can see less of it_.
+
+Everything on the web page
+stays exactly where it was
+relative to everything else --
+even the media-queries remain untouched --
+we're just looking at a smaller area
+of the overall page.
+
+**Scale factor** is adjusting
+_the size one viewport in relation to the other_.
+Specifically, the layout viewport
+can be scaled up larger (but not smaller)
+than the visual viewport.
+Since that happens _after rendering_,
+it has no impact on our page layout,
+or the available pixels,
+or any media queries.
+
+You might also notice a lack of scrollbars.
+We're not overflowing the box,
+we're zoomed in to view
+one smaller part of the box --
+and browsers handle that differently.
+
+### CSS `zoom` and `scale` properties
+
```

Comment:
The Scale vs Zoom demo could go either at the end of this previous section or the beginning here?